mmv1/api/type.go

```diff
@@ -912,6 +912,9 @@ func (t Type) UserProperties() []*Type {
 		}
 
 		return google.Reject(t.Properties, func(p *Type) bool {
+			if t.ResourceMetadata.IsTgcCompiler() {
```

Comment:
This change also affects `AllUserProperties`, as AllUserProperties = UserProperites + UserParameters
```
func (r Resource) AllUserProperties() []*Type {
	return google.Concat(r.UserProperites(), r.UserParameters())
}
```

`AllUserProperties` is also used to render the schemas for resources https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/tgc_next/services/resource.go.tmpl#L43

I think this is the reason for the test failure of `TestAccComputeBackendService_backendServiceExternalIapExample`.

Instead of modifying this function, we add the logic to the [flatten_property_method.go.tmpl](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/flatten_property_method.go.tmpl)

Reply:
We might still want to exclude output fields when making schemas for resources, since they won't be useful in either tfplan2cai or cai2hcl.